lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -848,6 +854,37 @@ bool SILPerformanceInliner::inlineCallsIntoFunction(SILFunction *Caller) {
   if (AppliesToInline.empty())
     return false;
 
+  // SWIFT_ENABLE_TENSORFLOW
+  // We should disable inlining for special functions if the any of following
+  // conditions hold:
+  //  - Is this a function in a TensorFlow module?
+  //  - Does this operate on Tensor values?
+  //
+  // Helper that returns true if function belongs to TensorFlow module.
+  auto isTensorFlowFunction = [](SILFunction *func) {
```

Comment:
adding such domain specific (tensor specific) knowledge to this "general purpose" mechanism of performance inliner does not feel ideal. This might prevent us from being able to upstream this patch later.

is there a way to refactor the code so that the caller / constructor of SILInliner can pass in a callback (or a bit) to decide whether `func` is eligible for inlining? The call-site from TF can then pass in a callback which does the "tensor op check" as written here.

Reply:
Yes, it is not ideal, but I prefer to leave this as is for now.

Reply:
I feel this is an important issue concering architecture ("layering") that we'll want to resolve ASAP. If you feel strongly about submitting this patch first, please go ahead, as long as we can follow up and address this one shortly.

Reply:
I agree with Mingsheng that this is a horrible hack, but I'm ok with it to unblock work - so long as there is a known path out of this.  What is the plan to resolve this?